lib/c-api/build.rs

```diff
@@ -79,6 +79,10 @@ fn build_wasm_c_api_headers(crate_dir: &str, out_dir: &str) {
 #if !defined(WASMER_WASM_H_MACROS)
 
 #define WASMER_WASM_H_MACROS
+
+// Include the deprecated API for import compatibility reasons
+#include "wasmer_deprecated.h"
```

Comment:
I'd suggest doing this the other way. Make users of the deprecated API `#include "wasmer_deprecated.h"` which in turn includes (and re-exports) the full `#include "wasmer.h"`. That way the deprecated doesn't need to redeclare any duplicate stuff, and people can easily build to the new API without using old parts by accident, and the new API is available to anyone using the old API to facilitate updating to the new API.

Reply:
We want to do it this way because I don't want to have breaking changes for people that are using the `#include "wasmer.h"` header for using the API.

That's the main reason for this. We will probably remove this on 2.0 :)